Initium-Odp/src/com/universeprojects/miniup/server/scripting/events/ScriptEvent.java

```diff
@@ -189,6 +187,35 @@ public boolean updatesGameState()
 		return jsResponse == JavascriptResponse.FullPageRefresh || reloadWidgets;
 	}
 	
+	private Map<Character, Set<String>> gameMessages = new HashMap<Character, Set<String>>();

```

Comment:
Might want to change this to a `HashMap<Character, List<String>>` since order actually matters here. This would only prevent the exact same message from being sent, which is highly unlikely (and should be prevented in script anyway).